packages/flutter_tools/test/general.shard/test/test_compiler_test.dart

```diff
@@ -113,6 +115,57 @@ void main() {
     ProcessManager: () => FakeProcessManager.any(),
     Logger: () => BufferLogger.test(),
   });
+
+  testUsingContext('TestCompiler updates generated_main.dart', () async {
+    final Directory fakeDartPlugin = fileSystem.directory('a_plugin');
+      fileSystem.file('pubspec.yaml').writeAsStringSync('''
+name: foo
+dependencies:
+  flutter:
+    sdk: flutter
+  a_plugin: 1.0.0
+''');
+      fileSystem.file('.packages').writeAsStringSync('a_plugin:/a_plugin/lib/');
+      fakeDartPlugin.childFile('pubspec.yaml')
+          ..createSync(recursive: true)
+          ..writeAsStringSync('''
+name: a_plugin
+flutter:
+  plugin:
+    implements: a
+    platforms:
+      linux:
+        dartPluginClass: APlugin
+environment:
+  sdk: ">=2.14.0 <3.0.0"
+  flutter: ">=2.5.0"
+''');
+
+    residentCompiler.compilerOutput = const CompilerOutput('abc.dill', 0, <Uri>[]);
+    final FakeTestCompiler testCompiler = FakeTestCompiler(
+      debugBuild,
+      FlutterProject.fromDirectoryTest(fileSystem.currentDirectory),
+      residentCompiler,
+    );
+
+    await testCompiler.compile(Uri.parse('test/foo.dart'));
+
+    final File generatedMain = fileSystem
+      .directory('.dart_tool')
+      .childDirectory('flutter_build')
+      .childFile('generated_main.dart');
+
+    expect(generatedMain, exists);
+    expect(
+      generatedMain.readAsStringSync(),
+      contains("import 'test/foo.dart' as entrypoint;")
```

Comment:
what if someone changes the generated code to `// import 'test/foo.dart' as entrypoint;`?  
could we ensure the entire output?